src/TagLib/Ogg/XiphComment.cs

```diff
@@ -211,6 +231,13 @@ public void SetField (string key, params string [] values)
 				field_list [key] = result.ToArray ();
 			else
 				field_list.Add (key, result.ToArray ());
+
+			// If the user is overwriting a picture field, reset
+			// the state in the pictures array
+			if (IsPictureField (key)) {
+				picture_fields_dirty = false;
+				pictures = null;
+			}
```

Comment:
This block of code is repeated a couple times and there are other places that reference similar code. Is it possible to refactor it out to a method? Maybe something along the lines of:

```void ResetPicturesState (string key) // maybe pass them in as parameters? ie, bool dirty = false, IPictures[] picture
{
    if (IsPictureField (key)) {
        pictures_fields_dirty = false;
        pictures = null;
    }
}```

Reply:
Fixed in 4c2aec942b4da9afb713f9d725948cdd2ebcdf8e